src/protocol/prf_sharding/mod.rs

```diff
@@ -117,14 +160,24 @@ impl InputsRequiredFromPrevRow {
             )
             .await?;
 
-        let attributed_trigger_value = zero_out_trigger_value_unless_attributed(
+        let mut attributed_trigger_value = zero_out_trigger_value_unless_attributed(
             ctx.narrow(&Step::AttributedTriggerValue),
             record_id,
             &did_trigger_get_attributed,
             &tv,
         )
         .await?;
 
+        attributed_trigger_value = zero_out_trigger_value_if_outside_attribution_window(
+            ctx.narrow(&Step::CheckAttributionWindow),
```

Comment:
I recommend saving a few multiplications by lifting out the "Is this within the attribution window" and storing that as a local variable. Then multiply that with `did_trigger_get_attributed`, then pass the result of that multiplication into a method that multiplies with all the bits of the trigger value.